crates/iota/src/client_commands.rs

```diff
@@ -552,16 +602,81 @@ pub enum IotaClientCommands {
     },
 }
 
-/// Global options for most transaction execution related commands
-#[derive(Args, Debug, Clone)]
-pub struct Opts {
-    /// An optional gas budget for this transaction (in NANOS). If gas budget is
+/// Arguments used to provide coins for a gas payment
+#[derive(Args, Debug, Default)]
+pub struct PaymentArgs {
+    /// IDs of gas objects to be used for gas payment. If none are provided,
+    /// coins are selected automatically to cover the gas budget.
+    #[arg(long, num_args(1..))]
+    pub gas: Vec<ObjectID>,
+}
+
+impl PaymentArgs {
+    /// Output the payment args as a vec of strings for CLI usage.
+    pub fn into_args(self) -> Vec<String> {
+        let mut args = Vec::new();
+        for gas_id in self.gas {
+            args.push(format!("--gas {}", gas_id));
```

Comment:
Since it's taking ownership of self, you could probably `into_iter().map()` so its aware of the capacity, would probably be more efficient. But I'm nitpicking a bit.